> rustresult Err:
RustlikeResult {
  _type: 'err',
  _value: undefined,
  _error: 'error message',
  _symbol: Symbol(rust_result)
}

> neverthrow err:
Err { error: 'error message' }

> effect Effect.fail:
{
  _id: 'Exit',
  _tag: 'Failure',
  cause: { _id: 'Cause', _tag: 'Fail', failure: 'error message' }
}

> effect Exit.fail:
{
  _id: 'Exit',
  _tag: 'Failure',
  cause: { _id: 'Cause', _tag: 'Fail', failure: 'error message' }
}

Loop N: 100,000
┌─────────┬──────────────────────┬──────────────────────┬───────────────────────┬────────────────┬───────────────┬─────────┐
│ (index) │ task                 │ mean (ns)            │ median (ns)           │ mean (op/s)    │ median (op/s) │ samples │
├─────────┼──────────────────────┼──────────────────────┼───────────────────────┼────────────────┼───────────────┼─────────┤
│ 0       │ 'rustresult Err'     │ '459830.16 ± 0.59%'  │ '442600.01'           │ '2213 ± 0.53%' │ '2259'        │ 2175    │
│ 1       │ 'neverthrow err'     │ '273295.22 ± 0.56%'  │ '252999.90 ± 0.12'    │ '3753 ± 0.48%' │ '3953'        │ 3660    │
│ 2       │ 'effect Effect.fail' │ '1179306.25 ± 0.62%' │ '1150750.04 ± 49.95'  │ '854 ± 0.55%'  │ '869'         │ 848     │
│ 3       │ 'effect Exit.fail'   │ '1169452.22 ± 0.67%' │ '1134950.04 ± 349.88' │ '863 ± 0.60%'  │ '881'         │ 856     │
└─────────┴──────────────────────┴──────────────────────┴───────────────────────┴────────────────┴───────────────┴─────────┘
